pkg/util/grpcconnection/config.go

```diff
@@ -151,14 +147,10 @@ func (c *ClientConfig) DialWithTimeOut(paths []string, timeout time.Duration) (*
 	return nil, utilerrors.NewAggregate(allErrs)
 }
 
-func createGRPCConnection(path string, timeout time.Duration, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-
-	cc, err := grpc.DialContext(ctx, path, opts...)
```

Comment:
Hi @RainbowMango , 

According to the documentation from [go-grpc](https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md), the `grpc.WithBloack()` and `grpc.DialContext` are deprecated, and it seems that using the timeout functionality in the manner we have might not be recommended. Instead, they suggest creating clients using `NewClient` rather than the deprecated Dial method, which aligns with best practices for managing client connections.

Additionally, I saw a relevant [PR](https://github.com/open-telemetry/opentelemetry-go/pull/5151) in the OpenTelemetry project, which also moved away from the timeout feature in favor of the new client creation method. Based on these findings, I am considering removing the timeout functionality from our implementation.

What do you think about this? 

Thank you for your guidance and support!

Reply:
cc the author @zhzhuang-zju to take a look

Reply:
@seanlaii Thank you for pointing out this issue. It can be confirmed that using `grpc.Dial` to create `ClientConn` is not the best practice. I will review the documentation you provided in detail and use it as a basis to review the current modifications.

Reply:
Compared to other modifications, the changes to the gRPC module will involve functionality. It might be better to submit them as a separate PR, which would allow for a more focused review. WDYT?

Reply:
Sounds good. I disable the syntax check for `grpc.DialContext` and `grpc.WithBlock` with some comments. We can discuss it in a separate PR. Thank you!